src/main/java/io/cdap/plugin/dynamo/source/DynamoDBBatchSourceConfig.java

```diff
@@ -0,0 +1,316 @@
+  }
+
+  public void setFilterQuery(@Nullable String filterQuery) {
+    this.filterQuery = filterQuery;
+  }
+
+  @Nullable
+  public String getNameMappings() {
+    return nameMappings;
+  }
+
+  public void setNameMappings(@Nullable String nameMappings) {
+    this.nameMappings = nameMappings;
+  }
+
+  public String getValueMappings() {
+    return valueMappings;
+  }
+
+  public void setValueMappings(String valueMappings) {
+    this.valueMappings = valueMappings;
+  }
+
+  public String getPlaceholderType() {
+    return placeholderType;
+  }
+
+  public void setPlaceholderType(String placeholderType) {
+    this.placeholderType = placeholderType;
+  }
+
+  @Nullable
+  public String getReadThroughput() {
+    return readThroughput;
+  }
+
+  public void setReadThroughput(@Nullable String readThroughput) {
+    this.readThroughput = readThroughput;
+  }
+
+  @Nullable
+  public String getReadThroughputPercentage() {
+    return readThroughputPercentage;
+  }
+
+  public void setReadThroughputPercentage(@Nullable String readThroughputPercentage) {
+    this.readThroughputPercentage = readThroughputPercentage;
+  }
+
+  public String getSchema() {
+    return schema;
+  }
+
+  public void setSchema(String schema) {
+    this.schema = schema;
+  }
+
+  public void validate(FailureCollector failureCollector) {
+    validateTableName(failureCollector);
+    validateSchema(failureCollector);
+  }
+
+  public Schema getParsedSchema() {
+    try {
+      return Schema.parseJson(schema);
+    } catch (IOException e) {
+      throw new IllegalStateException("Invalid schema: " + e.getMessage(), e);
+    }
+  }
+
+  /**
+   * Validates whether the table name follows the DynamoDB naming rules and conventions or not.
+   */
+  private void validateTableName(FailureCollector failureCollector) {
+    int tableNameLength = tableName.length();
+    if (tableNameLength < 3 || tableNameLength > 255) {
+      failureCollector.addFailure(
+        String.format("Table name '%s' does not follow the DynamoDB naming rules.", tableName),
+        "Table name must be between 3 and 255 characters long.")
+        .withConfigProperty(TABLE_NAME);
+    }
+
+    String pattern = "^[a-zA-Z0-9_.-]+$";
+    Pattern patternObj = Pattern.compile(pattern);
+
+    if (!Strings.isNullOrEmpty(tableName)) {
+      Matcher matcher = patternObj.matcher(tableName);
+      if (!matcher.find()) {
+        failureCollector.addFailure(
+          String.format("Table name '%s' does not follow the DynamoDB naming rules.", tableName),
+          "Table names can contain only the following characters: 'a-z, A-Z, 0-9, underscore(_), " +
+            "dot(.) and dash(-)'.")
+          .withConfigProperty(TABLE_NAME);
+
+      }
+    }
+  }
+
+  private void validateSchema(FailureCollector failureCollector) {
+    try {
+      Schema.parseJson(schema);
+    } catch (IOException e) {
+      failureCollector.addFailure("Invalid output schema: " + e.getMessage(), "Schema should be valid")
```

Comment:
use a null corrective action instead of 'Schema should be valid', since that doesn't give the user any useful information about how to fix it.